test-lib/test.js

```diff
@@ -1,57 +1,68 @@
 const fs = require('fs-extra');
 const path = require('path');
 
-const testNodeModules = path.join(__dirname, '/../test/node_modules');
-fs.removeSync(testNodeModules);
-fs.mkdirSync(testNodeModules);
-fs.symlinkSync(path.join(__dirname, '/..'), path.join(testNodeModules, 'apostrophe'), 'dir');
-
-const extras = path.join(__dirname, '../test/extra_node_modules');
-const dirs = fs.existsSync(extras) ? fs.readdirSync(extras) : [];
-for (const dir of dirs) {
-  fs.symlinkSync(path.join(extras, dir), path.join(testNodeModules, dir), 'dir');
-}
-
-// Need a "project level" package.json for functionality that checks
-// whether packages in node_modules are project level or not
-
-const packageJson = path.join(__dirname, '/../test/package.json');
-// Remove it first, in case it's the old-style symlink to the main package.json,
-// which would break
-fs.removeSync(packageJson);
-const packageJsonInfo = {
-  name: 'test',
-  dependencies: {
-    apostrophe: '^3.0.0'
-  },
-  devDependencies: {
-    'test-bundle': '1.0.0'
+const setupPackages = ({
+  folder = 'test',
+  type = 'commonjs'
+}) => {
+  const testNodeModules = path.join(__dirname, '../', folder, 'node_modules/');
+  fs.removeSync(testNodeModules);
+  fs.mkdirSync(testNodeModules);
+  fs.symlinkSync(path.join(__dirname, '../'), path.join(testNodeModules, 'apostrophe'), 'dir');
+
+  const extras = path.join(__dirname, '../', folder, 'extra_node_modules/');
+  const dirs = fs.existsSync(extras) ? fs.readdirSync(extras) : [];
+  for (const dir of dirs) {
+    fs.symlinkSync(path.join(extras, dir), path.join(testNodeModules, dir), 'dir');
   }
-};
-for (const dir of dirs) {
-  // Add namespaced modules support
-  if (dir.startsWith('@')) {
-    const submodules = fs.readdirSync(path.join(extras, dir));
-    for (const submodule of submodules) {
-      packageJsonInfo.dependencies[`${dir}/${submodule}`] = '1.0.0';
+
+  // Need a "project level" package.json for functionality that checks
+  // whether packages in node_modules are project level or not
+
+  const packageJson = path.join(__dirname, '../', folder, 'package.json');
+  // Remove it first, in case it's the old-style symlink to the main package.json,
+  // which would break
+  fs.removeSync(packageJson);
+  const packageJsonInfo = {
+    name: folder,
+    // type, // prevent asset build with esm support
```

Comment:
only way to make assets test passes

Reply:
What is your concern about the change?

Reply:
Actually this is something I've added, can't be removed. No issues there.